.github/workflows/docs-gen-and-push.yaml

```diff
@@ -0,0 +1,55 @@
+name: Generate and push docs
+
+on:
+  push:
+    branches:
+      - main
+      - "release-*"
```

Comment:
is this correct to have so many branches?

Reply:
My idea was to have versioned docs if we intend to do so. We can leave it to just `main` for now.